### ISSUE TITLE
FileSystemModification Activities are generated too late

### ISSUE THREAD

**Issue by dpp-gerrit** [saros-project/saros#10]
Hi,

when the session host renames a package, a error message appears on every client that has this package shown in the package explorer. When the package's parent node is collapsed, the error disappears.

```
eclipse.buildId=unknown
java.version=1.6.0_19
java.vendor=Sun Microsystems Inc.
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=de_DE
Framework arguments: -product org.eclipse.platform.ide
Command-line arguments: -product org.eclipse.platform.ide -data C:\Users\Christophe\workspace/../workspace-alice-host -dev file:C:/Users/Christophe/workspace/.metadata/.plugins/org.eclipse.pde.core/Saros_Alice_Host/dev.properties -os win32 -ws win32 -arch x86_64

This is a continuation of log file C:\Users\Christophe\workspace-alice-host\.metadata\.bak_0.log
Created Time: 2010-06-24 15:46:14.075

Warning
Thu Jun 24 16:03:08 CEST 2010
Viewport for editor which was never activated: SPath [editorType=txt, path=src/test234/pkg_xyz345/TestXYZ.java, project=Pupsiprojekt 3] (RemoteEditorManager.java:107)
```

Reported by: csokol

Original Ticket: [dpp/bugs/371](https://sourceforge.net/p/dpp/bugs/371)

**Comment by fzieris** [comment]
I tried to reproduce this old issue and it still exists.

* Alice invites Bob with project "Test" and a single package (named "trumsbums")
* Bob joins the session and starts to follow Alice.
* Both have the same file opened: trumsbums/Bums.java
* Alice renames the package to "ums"
* Bob leaves the Follow Mode, his editor of "Bums.java" is closed, and his log file has the following messages (full logs are attached)
```
WARN  18:00:23,938 [main] (UserEditorState.java:56) Viewport for editor which was never activated: SPath [project=P/Test (EclipseProjectImpl), projectRelativePath=src/ums/Bums.java]
[...]
TRACE 18:00:24,067 [dpp-activity-sender] (ActivitySequencer.java:535) send (003) dev7_alice_stf@saros-con.imp.fu-berlin.de/Saros -> [StopFollowingActivity(dev7_bob_stf@saros-con.imp.fu-berlin.de/Saros), StopFollowingActivity(dev7_bob_stf@saros-con.imp.fu-berlin.de/Saros), EditorActivity(type: ACTIVATED, path: null)]
[...]
ERROR 18:00:24,157 [main] (RemoteWriteAccessManager.java:147) Attempting to connect to file which is not available locally: SPath [project=P/Test (EclipseProjectImpl), projectRelativePath=src/ums/Bums.java]
StackTrace:
	at de.fu_berlin.inf.dpp.editor.RemoteWriteAccessManager.connectDocumentProvider(RemoteWriteAccessManager.java:148)
	at de.fu_berlin.inf.dpp.editor.RemoteWriteAccessManager.updateConnectionState(RemoteWriteAccessManager.java:195)
	at de.fu_berlin.inf.dpp.editor.RemoteWriteAccessManager.receive(RemoteWriteAccessManager.java:88)
	at de.fu_berlin.inf.dpp.activities.EditorActivity.dispatch(EditorActivity.java:115)
	at de.fu_berlin.inf.dpp.session.AbstractActivityConsumer.exec(AbstractActivityConsumer.java:85)
	at de.fu_berlin.inf.dpp.editor.EditorManager$2.exec(EditorManager.java:192)
	at de.fu_berlin.inf.dpp.session.internal.SarosSession$2.execute(SarosSession.java:191)
	at de.fu_berlin.inf.dpp.session.internal.ActivityHandler$3.run(ActivityHandler.java:369)
	at de.fu_berlin.inf.dpp.util.ThreadUtils$1.run(ThreadUtils.java:38)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:35)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:135)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4140)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3757)
	at org.eclipse.ui.internal.Workbench.runEventLoop(Workbench.java:2701)
	at org.eclipse.ui.internal.Workbench.runUI(Workbench.java:2665)
	at org.eclipse.ui.internal.Workbench.access$4(Workbench.java:2499)
	at org.eclipse.ui.internal.Workbench$7.run(Workbench.java:679)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:668)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:149)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:123)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:110)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:79)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:344)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:179)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:622)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:577)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1410)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1386)
TRACE 18:00:24,158 [main] (SharedProjectFileDecorator.java:149) remote user: dev7_alice_stf@saros-con.imp.fu-berlin.de/Saros activated an editor -> SPath [project=P/Test (EclipseProjectImpl), projectRelativePath=src/ums/Bums.java]
WARN  18:00:24,159 [main] (SharedProjectFileDecorator.java:162) resource for editor SPath [project=P/Test (EclipseProjectImpl), projectRelativePath=src/ums/Bums.java] does not exist locally
TRACE 18:00:24,159 [main] (RemoteWriteAccessManager.java:179) .updateConnectionState(SPath [project=P/Test (EclipseProjectImpl), projectRelativePath=src/ums/Bums.java])
TRACE 18:00:24,159 [main] (RemoteWriteAccessManager.java:193) .updateConnectionState File SPath [project=P/Test (EclipseProjectImpl), projectRelativePath=src/ums/Bums.java] will be connected 
ERROR 18:00:24,160 [main] (RemoteWriteAccessManager.java:147) Attempting to connect to file which is not available locally: SPath [project=P/Test (EclipseProjectImpl), projectRelativePath=src/ums/Bums.java]
StackTrace:
	at de.fu_berlin.inf.dpp.editor.RemoteWriteAccessManager.connectDocumentProvider(RemoteWriteAccessManager.java:148)
	at de.fu_berlin.inf.dpp.editor.RemoteWriteAccessManager.updateConnectionState(RemoteWriteAccessManager.java:195)
	at de.fu_berlin.inf.dpp.editor.RemoteWriteAccessManager.receive(RemoteWriteAccessManager.java:88)
	at de.fu_berlin.inf.dpp.activities.EditorActivity.dispatch(EditorActivity.java:115)
	at de.fu_berlin.inf.dpp.session.AbstractActivityConsumer.exec(AbstractActivityConsumer.java:85)
	at de.fu_berlin.inf.dpp.editor.EditorManager$2.exec(EditorManager.java:192)
	at de.fu_berlin.inf.dpp.session.internal.SarosSession$2.execute(SarosSession.java:191)
	at de.fu_berlin.inf.dpp.session.internal.ActivityHandler$3.run(ActivityHandler.java:369)
	at de.fu_berlin.inf.dpp.util.ThreadUtils$1.run(ThreadUtils.java:38)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:35)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:135)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4140)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3757)
	at org.eclipse.ui.internal.Workbench.runEventLoop(Workbench.java:2701)
	at org.eclipse.ui.internal.Workbench.runUI(Workbench.java:2665)
	at org.eclipse.ui.internal.Workbench.access$4(Workbench.java:2499)
	at org.eclipse.ui.internal.Workbench$7.run(Workbench.java:679)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:668)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:149)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:123)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:110)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:79)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:344)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:179)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:622)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:577)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1410)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1386)
TRACE 18:00:24,160 [main] (SharedProjectFileDecorator.java:149) remote user: dev7_alice_stf@saros-con.imp.fu-berlin.de/Saros activated an editor -> SPath [project=P/Test (EclipseProjectImpl), projectRelativePath=src/ums/Bums.java]
WARN  18:00:24,161 [main] (SharedProjectFileDecorator.java:162) resource for editor SPath [project=P/Test (EclipseProjectImpl), projectRelativePath=src/ums/Bums.java] does not exist locally
```


[alice_log.txt](https://github.com/saros-project/saros/files/1852801/alice_log.txt)
[bob_log.txt](https://github.com/saros-project/saros/files/1852803/bob_log.txt)

**Comment by srossbach** [comment]
I tracked this down. While I could apply a fix that may work on very fast machines I do not see the point of doing it.

The general problem in Eclipse is that we do get notified about file changes in the background most of times (and also only after they were already performed). Meanwhile the main thread continues and starts to fire Editor Activities for files where the generation of the File Activities is still in progress.

**Comment by srossbach** [comment]
While I get your intention of closing this I do not think it is ok to close this without a final statement. The impact is pretty huge closing this issue with: will not fix.

**Comment by tobous** [comment]
> While I get your intention of closing this I do not think it is ok to close this without a final statement. The impact is pretty huge closing this issue with: will not fix.

I basically read your previous statement as "nothing we can do about it". And, as the issue was marked as "wontfix", I though it was ok to closed it. But if you think that there is still something to be done here, feel free to reopen it. (I also closed another issue you marked as "wontfix". You might want to have a look at that one as well.)
